### PR TITLE
#2248 New limiter block based on min and max functions

### PR DESCRIPTION
**Pull request by ErwanGuichard49**

### CODE REVIEW

File: dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/Limiter.mo [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+model Limiter
+  import Modelica;
+
+  extends Modelica.Blocks.Interfaces.SISO;
+
+  parameter Real uMax "Upper limit of input signal";
+  parameter Real uMin= -uMax "Lower limit of input signal";
+
+equation
+  y = smooth(0, min(uMax, max(uMin, u)));
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/Limiter.mo:25]: I don't understand why you are using smooth(0,


I would have expected something like y = min(uMax, max(uMin, u)); Is there a reason ?

File: dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/Limiter.mo [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite
+* of simulation tools for power systems.
+*/
+
+model Limiter
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/Limiter.mo:16]: Comment

File: dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/Limiter.mo [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  extends Modelica.Blocks.Interfaces.SISO;
+
+  parameter Real uMax "Upper limit of input signal";
+  parameter Real uMin= -uMax "Lower limit of input signal";
+
+equation
+  y = smooth(0, min(uMax, max(uMin, u)));
+
+end Limiter;
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/Limiter.mo:27]: Documentation + graphical view (like in the MSL)

**Review by cossartque — CHANGES_REQUESTED** [review]

File: dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/Limiter.mo (hunk 1 above)
Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/Limiter.mo:25]: I followed the example set by noEvent() documentation. I will try without smooth.

Comment (ErwanGuichard49) [reply dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/Limiter.mo]: It works well without smooth, with no impact on the results.

**Comment by ErwanGuichard49** [comment]
This new block is no longer necessary to ExTkjEdf1.

**Pull request closed without merging**